src/parser/datetime.rs

```diff
@@ -68,6 +70,92 @@ pub(crate) fn tokenize_interval(value: &str) -> Result<Vec<IntervalToken>, Parse
     Ok(toks)
 }
 
+fn tokenize_timezone(value: &str) -> Result<Vec<IntervalToken>, ParserError> {
+    let mut toks: Vec<IntervalToken> = vec![];
+    let mut num_buf = String::with_capacity(4);
+    /* If the timezone string has a colon, we need to parse all numbers naively, otherwise we need to parse long sequences of digits as [..hhhhmm] */
+    let split_nums: bool = value.find(':').unwrap_or(0) == 0;
+
+    fn parse_num(
+        toks: &mut Vec<IntervalToken>,
+        n: &str,
+        split_nums: bool,
+        idx: usize,
+    ) -> Result<(), ParserError> {
+        if n.is_empty() {
+            return Ok(());
+        }
+
+        if n.len() <= 2 || split_nums == false {
```

Comment:
I kind of feel like this `if` would be easier to read if it just set up first and second unconditionally and then parsed:

```
let (first, second) = if n.len() > 2 || split_nums {
    let (first, second) = n.split_at(n.len() -2)
    (first, Some(second))
} else {
    (n, None)
};
toks.push(IntervalToken::Num(..)?);
if let Some(second) = second {
    toks.push(IntervalToken::Num(..)?);
}
```

Reply:
ah ok I wasn't really aware of that syntax / pattern - I'll clean up and do something similar in a few other places too

Reply:
changed this here